msdk-io/msdk-io-netcdf/src/main/java/io/github/msdk/io/netcdf/NetCDFFileExportMethod.java

```diff
@@ -0,0 +1,198 @@
+          writer.addDimension(null, "point_number", scanStartPositions[totalScans]);
+      List<Dimension> pointNumValDims = new ArrayList<>();
+      pointNumValDims.add(pointNumDim);
+
+      // populate the mass values
+      Array massValueArray =
+          Array.factory(double.class, new int[] {scanStartPositions[totalScans]});
+      idx = 0;
+      for (MsScan scan : scans) {
+        double mzValues[] = scan.getMzValues();
+        if (canceled) {
+          writer.close();
+          return null;
+        }
+
+        for (int i = 0; i < mzValues.length; i++)
+          massValueArray.setDouble(idx++, mzValues[i]);
+      }
+
+      // Write the mass values
+      Variable massValueVariable =
+          writer.addVariable(null, "mass_values", DataType.FLOAT, pointNumValDims);
+      massValueVariable.addAttribute(new Attribute("units", "M/Z"));
+      massValueVariable.addAttribute(new Attribute("scale_factor", massValueScaleFactor));
+
+      progress = 0.5f;
+
+      // populate the intensity values
+      Array intensityValueArray =
+          Array.factory(double.class, new int[] {scanStartPositions[totalScans]});
+      idx = 0;
+      for (MsScan scan : scans) {
+        if (canceled) {
+          writer.close();
+          return null;
+        }
+        float intensityValues[] = scan.getIntensityValues();
+        for (int i = 0; i < intensityValues.length; i++)
+          intensityValueArray.setDouble(idx++, intensityValues[i]);
+      }
+
+      // Write the intensity values
+      Variable intensityValueVariable =
+          writer.addVariable(null, "intensity_values", DataType.FLOAT, pointNumValDims);
+      intensityValueVariable.addAttribute(new Attribute("units", "Arbitrary Intensity Units"));
+      intensityValueVariable.addAttribute(new Attribute("scale_factor", intensityValueScaleFactor));
+
+      progress = 0.75f;
+
+      // Populate scan times
+      Array scanTimeArray = Array.factory(float.class, new int[] {totalScans});
+      idx = 0;
+      for (MsScan scan : scans)
+        scanTimeArray.setFloat(idx++, scan.getRetentionTime());
+
+      if (canceled) {
+        writer.close();
+        return null;
+      }
+
+      // Create the scan times variable
+      Variable scanTimeVariable =
+          writer.addVariable(null, "scan_acquisition_time", DataType.FLOAT, scanIndexDims);
+
+      // Create the netCDF-3 file
+      writer.create();
+
+      // Write out the non-record variables
+      writer.write(scanIndexVariable, scanIndexArray);
+      writer.write(massValueVariable, massValueArray);
+      writer.write(intensityValueVariable, intensityValueArray);
+      writer.write(scanTimeVariable, scanTimeArray);
+
+      // Close the writer
+      writer.close();
+
+      progress = 1.0f;
+
+    } catch (IOException | InvalidRangeException e) {
+      new MSDKException(e);
+    }
+
+    return null;
+  }
+
+  @Override
+  public Float getFinishedPercentage() {
+    return progress;
```

Comment:
Why not calculate the progress from the number of finished scans, same as in the mzML exporter?